src/bin/s3Purge.ts

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env node
+
+import { Command } from 'commander'
+import { LIB_VERSION } from '../version'
+import { S3Scan, S3_SCAN_HTTPS_AGENT } from '../lib'
+
+const program = new Command()
+
+program
+  .version(LIB_VERSION)
+  .description(
+    `A tool Provides a writable stream that expects you to write line-delimited S3 keys
+    into it, and performs an S3.deleteObject request on each key`,
+  )
+  .usage('Usage: s3purge <s3url> [--concurrency=num] [--dryrun] [--quiet]')
+  .argument('<s3Url>', 'an S3 uri of the type s3://bucket/prefix')
+  .option('-d, --dryrun [boolean]', 'Simulate the deletion only', false)
+  .option(
+    '-q, --quiet  [boolean]',
+    'Boolean value to run in silent mode',
+    false,
+  )
+  .option('-c, --concurrency  [number]', 'Concurrency to use')
+  .parse(process.argv)
+
+const options = { agent: S3_SCAN_HTTPS_AGENT, body: true, ...program.opts() }
+const s3Url = program.args[0]
+const { quiet, dryrun } = program.opts()
+const interval: NodeJS.Timeout | undefined =
+  !quiet && !dryrun
+    ? setInterval(function () {
+        process.stdout.write(
+          `[KDeleted ${purge.deletedCount} @ ${purge.rate()}/s`,
+        )
+      }, 500)
+    : undefined
+
+const purge = S3Scan.Purge(s3Url, options, err => {
+  if (!quiet) clearInterval(interval)
+  setTimeout(() => {
+    console.log(quiet ? purge.deletedCount : '')
```

Comment:
I think the condition here is wrong. Currently, when `quiet=true`, the deleted count is printed and otherwise an empty line is printed. I believe this should be the reverse.

Moreover, instead of logging an empty string, I think it'd be better to not log anything at all when the `quiet` flag is set.

```suggestion
    if (!quiet) {
      console.log(purge.deletedCount);
    }
```